contracts/src/utils/TokenScalingUtils.sol

```diff
@@ -0,0 +1,41 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+pragma solidity 0.8.18;
+
+library TokenScalingUtils {
+    function applyTokenScale(
+        uint256 tokenMultiplier,
+        bool multiplyOnDestination,
+        uint256 amount
+    ) internal pure returns (uint256) {
+        return _scaleTokens(tokenMultiplier, multiplyOnDestination, amount, true);
+    }
+
+    function removeTokenScale(
+        uint256 tokenMultiplier,
+        bool multiplyOnDestination,
+        uint256 amount
+    ) internal pure returns (uint256) {
+        return _scaleTokens(tokenMultiplier, multiplyOnDestination, amount, false);
+    }
```

Comment:
Can we leave comments on these as to which should be used for send/receive functions on the source?